pufaAgent/getlogs_Pufa.py

```diff
@@ -0,0 +1,220 @@
+
+
+def is_time_format(time_string, datetime_format):
+    """
+    Determines the validity of the input date-time string according to the given format
+    Parameters:
+    - `timeString` : datetime string to check validity
+    - `temp_id` : datetime format to compare with
+    """
+    try:
+        datetime.strptime(str(time_string), datetime_format)
+        return True
+    except ValueError:
+        return False
+
+
+def get_timestamp_for_zone(date_string, time_zone, datetime_format):
+    dtexif = datetime.strptime(date_string, datetime_format)
+    tz = pytz.timezone(time_zone)
+    tztime = tz.localize(dtexif)
+    epoch = long((tztime - datetime(1970, 1, 1, tzinfo=pytz.utc)).total_seconds()) * 1000
+    return epoch
+
+def getAgentConfigVars():
+    configVars = {}
+    try:
+        if os.path.exists(os.path.join(parameters['homepath'], "pufaAgent", "config.ini")):
+            parser = SafeConfigParser()
+            parser.read(os.path.join(parameters['homepath'], "pufaAgent", "config.ini"))
+            tag = parser.get('pufa', 'tag')
+            fileName = parser.get('pufa', 'file_name')
+            licenseKey = parser.get('pufa', 'insightFinder_license_key')
+            projectName = parser.get('pufa', 'insightFinder_project_name')
+            userName = parser.get('pufa', 'insightFinder_user_name')
+            samplingInterval = parser.get('pufa', 'sampling_interval')
+            if len(tag) == 0:
+                logger.error("Agent not correctly configured(tag name). Check config file.")
+                sys.exit(1)
+            if len(fileName) == 0:
+                logger.error("Agent not correctly configured(file name). Check config file.")
+                sys.exit(1)
+            if len(licenseKey) == 0:
+                logger.error("Agent not correctly configured(license key). Check config file.")
+                sys.exit(1)
+            if len(projectName) == 0:
+                logger.error("Agent not correctly configured(project name). Check config file.")
+                sys.exit(1)
+            if len(userName) == 0:
+                logger.error("Agent not correctly configured(user name). Check config file.")
+                sys.exit(1)
+            if len(samplingInterval) == 0:
+                logger.error("Agent not correctly configured(sampling interval). Check config file.")
+                sys.exit(1)
+            configVars['tag'] = tag
+            configVars['fileName'] = fileName
+            configVars['licenseKey'] = licenseKey
+            configVars['projectName'] = projectName
+            configVars['userName'] = userName
+            if samplingInterval[-1:] == 's':
+                configVars['samplingInterval'] = float(float(samplingInterval[:-1]) / 60.0)
+            else:
+                configVars['samplingInterval'] = int(samplingInterval)
+    except IOError:
+        logger.error("config.ini file is missing")
+    return configVars
+
+
+def parseMessage():
```

Comment:
This function is reading messages for a file. Use a suggestive name or add comments